v5/pivotal/aggregator.go

```diff
@@ -0,0 +1,218 @@
+func (a *Aggregation) CommentsOfStories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.CommentsOfStory(projectID, storyID)
+	}
+	return a
+}
+
+// ReviewsOfStory adds multiple requests for getting the reviews of a story.
+func (a *Aggregation) ReviewsOfStory(projectID, storyID int) *Aggregation {
+	a.requests = append(a.requests, buildReviewsURL(projectID, storyID))
+	return a
+}
+
+// ReviewsOfStories adds multiple requests for getting the reviews of multiple stories.
+func (a *Aggregation) ReviewsOfStories(projectID int, storyIDs []int) *Aggregation {
+	for _, storyID := range storyIDs {
+		a.ReviewsOfStory(projectID, storyID)
+	}
+	return a
+}
+
+func buildStoryURLOnlyUsingStoryID(storyID int) string {
+	return fmt.Sprintf("/services/v5/stories/%d", storyID)
+}
+
+func buildStoryURL(projectID, storyID int) string {
+	return fmt.Sprintf("/services/v5/projects/%d/stories/%d", projectID, storyID)
+}
+
+func buildCommentsURL(projectID, storyID int) string {
+	return fmt.Sprintf("/services/v5/projects/%d/stories/%d/comments", projectID, storyID)
+}
+
+func buildReviewsURL(projectID, storyID int) string {
+	return fmt.Sprintf("/services/v5/projects/%d/stories/%d/reviews?fields=id,story_id,review_type,review_type_id,reviewer_id,status,created_at,updated_at,kind", projectID, storyID)
+}
+
+func maxPagesPagination(total int, perPage int) int {
+	pagesNumber := math.Ceil(float64(total) / float64(perPage))
+	return int(pagesNumber)
+}
+
+func paginate(reqs []string, currentPage, total int, perPage int) []string {
+	firstEntry := (currentPage - 1) * perPage
+	lastEntry := firstEntry + perPage
+
+	if lastEntry > total {
+		lastEntry = total
+	}
+
+	return reqs[firstEntry:lastEntry]
+}
+
+// Send completes the aggregation and sends it to Pivotal Tracker.
+func (a *Aggregation) Send() (*Aggregation, error) {
+	N := len(a.requests)
+	max := maxPagesPagination(N, requestsPerAggregation)
+	for currentPage := 1; currentPage <= max; currentPage++ {
+		currentReqs := paginate(a.requests, currentPage, N, requestsPerAggregation)
+
+		aggregatedResponse := make(map[string]interface{})
+
+		req, err := a.service.client.NewRequest("POST", aggregatorURL, currentReqs)
+		if err != nil {
+			return nil, err
+		}
+
+		_, err = a.service.client.Do(req, &aggregatedResponse)
+		if err != nil {
+			return nil, err
+		}
+
+		// Appending the response body into the current aggregation for getters.
+		for url, response := range aggregatedResponse {
+			a.aggregatedResponse[url] = response
+		}
+	}
+
+	return a, nil
+}
+
+// GetStoryOnlyUsingStoryID returns the story using only the story ID.
+func (a *Aggregation) GetStoryOnlyUsingStoryID(storyID int) (*Story, error) {
+	u := buildStoryURLOnlyUsingStoryID(storyID)
+	response, ok := a.aggregatedResponse[u]
+	if !ok {
+		return nil, fmt.Errorf("Story %d doesn't exist.", storyID)
+	}
+	byteData, _ := json.Marshal(response)
+
```

Comment:
We should return an error if this fails. Same with the other `json.Marshal` calls.

I wonder if it would make sense to wrap this in a helper function, e.g.:

```
func mapTo(from interface{}, to interface{}) error {
    b, err := json.Marshal(from)
    if err != nil {
         return err
    }
    return json.Unmarshal(b, &to)
}
```

This should reduce the repeated code and in the future might make it easier to replace this with something more efficient, e.g., https://github.com/mitchellh/mapstructure.